File: DSA/Navneetpant_590011569/Day_33/Q2.java
import java.util.*;

class Solution {
    public String postfixToPrefix(String exp) {
        Stack<String> st = new Stack<>();

        for (char ch : exp.toCharArray()) {
            if (Character.isLetterOrDigit(ch)) {
                st.push(String.valueOf(ch));
            } else {
                String op2 = st.pop();
                String op1 = st.pop();

                String temp = ch + op1 + op2;
                st.push(temp);
            }
        }

        return st.pop();
    }
}

File: DSA/Aarush_590017879/Day_33/Q2.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

string postfixToPrefix(string s) {
    stack<string> st;

    for (char ch : s) {

        if (ch >= 'A' && ch <= 'Z') {
            st.push(string(1, ch));
        }
        else {
            string op2 = st.top();
            st.pop();

            string op1 = st.top();
            st.pop();

            string temp = ch + op1 + op2;

            st.push(temp);
        }
    }

    return st.top();
}

int main() {
    string s = "ABC/-AK/L-*";

    cout << postfixToPrefix(s);

    return 0;
}